tests/src/python/test_qgsrasterblock.py

```diff
@@ -0,0 +1,58 @@
+"""QGIS Unit tests for QgsRasterBlock.
+
+From build dir, run:
+ctest -R PyQgsQgsRasterBlock -V
+
+.. note:: This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+"""
+
+__author__ = 'Till Frankenbach'
+__date__ = '24/07/2024'
+__copyright__ = 'Copyright 2024, The QGIS Project'
+
+import numpy
+
+from qgis.core import (
+    QgsRasterLayer,
+)
+import unittest
+from qgis.testing import start_app, QgisTestCase
+from qgis.testing.mocked import get_iface
+from qgis.PyQt.QtCore import QTemporaryDir
+
+from test_qgsrasterattributetable import createTestRasters
+
+# Convenience instances in case you may need them
+# not used in this test
+start_app()
+
+
+class TestQgsRasterBlock(QgisTestCase):
```

Comment:
This test needs to be added to the Cmake file too (https://github.com/qgis/QGIS/blob/master/tests/src/python/CMakeLists.txt#L264), otherwise it won't be run